src/s1ap/handlers/s1ap_encoder.c

```diff
@@ -1653,11 +1653,14 @@ int s1ap_mme_encode_erab_setup_request(struct erabsu_ctx_req_Q_msg *s1apPDU,
 
 	uint32_t transport_layer_address = htonl(s1apPDU->erab_su_list.erab_su_item[i].transportLayerAddress);
 	erab_to_be_setup->transportLayerAddress.size = 4;
-	erab_to_be_setup->transportLayerAddress.buf = (uint8_t*)&transport_layer_address;
+	erab_to_be_setup->transportLayerAddress.buf = calloc(4, sizeof(uint8_t));
```

Comment:
who is taking care of freeing this allocation.

Reply:
Taken care in https://github.com/omec-project/Nucleus/pull/83/commits/6a8fc5538302ca9fd2730d187bbf95268309ee01